wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -1151,12 +1151,28 @@ private AnnotationSpec wireFieldAnnotation(NameAllocator nameAllocator, Field fi
       result.addMember("adapter", "$S", adapterString(field.getType()));
     }
 
-    if (!field.isOptional()) {
-      if (field.isPacked()) {
-        result.addMember("label", "$T.PACKED", WireField.Label.class);
-      } else if (field.getLabel() != null) {
-        result.addMember("label", "$T.$L", WireField.Label.class, field.getLabel());
-      }
+    WireField.Label wireFieldLabel;
+    //noinspection ConstantConditions
+    switch (field.getEncodeMode()) {
+      case THROW_IF_ABSENT:
```

Comment:
should we rename `THROW_IF_ABSENT` to `REQUIRED` ? Fewer names is better

Reply:
fixed